File: frontend-mealsonwheels/src/types/foodSafety/FoodSafetyRecordDTO.ts
// src/types/management/FoodSafetyRecordDTO.ts

export interface FoodSafetyRecordDTO {
  id: number;
  inspectionDate: string; // ISO date-time string (e.g., "2025-06-06T14:00:00")
  safetyChecklistCompleted: boolean;
  inspectionNotes?: string;
  ingredientId: number;
}

File: backend-mealsonwheels/src/main/java/com/merrymeal/mealsonwheels/dto copy/FoodSafetyRecordDTO.java
package com.merrymeal.mealsonwheels_backend.dto;

import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class FoodSafetyRecordDTO {

    private Long id;

    @NotNull(message = "Inspection date is required")
    private LocalDateTime inspectionDate;

    @NotNull(message = "Safety checklist completion status is required")
    private Boolean safetyChecklistCompleted;

    private String inspectionNotes;

    @NotNull(message = "Ingredient ID is required")
    private Long ingredientId;
}
